Compilation failed at libtiff build

Windows 10 LTSC 64 bit, Intel i5 2500k, 16GB RAM DDR3 1600, GTX 1060 6GB (if it matters). I have Cygwin installed and bin in $PATH, don't know if that causes any trouble and also no WSL installed

[logs.zip](https://0x0.st/zZYN.zip)

```        Starting 64bit compilation of all tools
Imported options from ffmpeg_options.txt
Imported custom options from ffmpeg_options.txt
License: nonfree
Imported options from mpv_options.txt

        Starting 64bit compilation of global tools
18:08:47 ├ Running download_rustup...

[snip]

18:35:22 Installing libjpeg-turbo... done
18:35:26   Running git clone for libtiff...
18:35:46 ┌ libtiff git .............................. [Recently updated]
18:35:46 ├ Running cmake...
18:36:10 ├ Running build...
Likely error:
          ^~~~
ninja: build stopped: subcommand failed.
[1/6] Linking C executable contrib\addtiffo\addtiffo.exe
[2/6] Linking C executable contrib\dbs\tiff-grayscale.exe
FAILED: contrib/dbs/tiff-grayscale.exe
cmd.exe /C "cd . && D:\media-autobuild_suite-master\msys64\mingw64\bin\gcc.exe -mthreads -mtune=generic -O2 -pipe -DLIBXML_STATIC -Wall -Winline -W -Wformat-security -Wpointer-arith -Wdisabled-optimization -Wno-unknown-pragmas -Wdeclaration-after-statement -fstrict-aliasing -O3 -DNDEBUG  -pipe -static-libgcc -static-libstdc++ contrib/dbs/CMakeFiles/tiff-grayscale.dir/tiff-grayscale.c.obj  -o contrib\dbs\tiff-grayscale.exe -Wl,--major-image-version,0,--minor-image-version,0  libtiff/libtiff.a port/libport.a C:/tools/cygwin64/lib/libm.a -Wl,-Bstatic -lz -ljpeg -llzma -lzstd -Wl,-Bdynamic -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmsvcrt.a(desis01151.o):(.text+0x0): multiple definition of `log10'
C:/tools/cygwin64/lib/libm.a(t-d000863.o):fake:(.text+0x0): first defined here
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
build failed. Check D:/media-autobuild_suite-master/build/libtiff-git/build-64bit/ab-suite.build.log
This is required for other packages, so this script will exit.
18:36:42   Creating diagnostics file...```

That's due to a conflict with msvc's libs and mingw.
I do have a patch for it, but was unsure if it was needed.

On Tue, Apr 9, 2019 at 12:33 PM Hackerpcs <notifications@github.com> wrote:

> Windows 10 LTSC 64 bit, Intel i5 2500k, GTX 1060 6GB (if it matters)
>
> logs.zip <https://0x0.st/zZYN.zip>
>
> Imported options from ffmpeg_options.txt
>
> Imported custom options from ffmpeg_options.txt
>
> License: nonfree
>
> Imported options from mpv_options.txt
>
>
>
>         Starting 64bit compilation of global tools
>
> 18:08:47 ├ Running download_rustup...
>
>
>
> [snip]
>
>
>
> 18:35:22 Installing libjpeg-turbo... done
>
> 18:35:26   Running git clone for libtiff...
>
> 18:35:46 ┌ libtiff git .............................. [Recently updated]
>
> 18:35:46 ├ Running cmake...
>
> 18:36:10 ├ Running build...
>
> Likely error:
>
>           ^~~~
>
> ninja: build stopped: subcommand failed.
>
> [1/6] Linking C executable contrib\addtiffo\addtiffo.exe
>
> [2/6] Linking C executable contrib\dbs\tiff-grayscale.exe
>
> FAILED: contrib/dbs/tiff-grayscale.exe
>
> cmd.exe /C "cd . && D:\media-autobuild_suite-master\msys64\mingw64\bin\gcc.exe -mthreads -mtune=generic -O2 -pipe -DLIBXML_STATIC -Wall -Winline -W -Wformat-security -Wpointer-arith -Wdisabled-optimization -Wno-unknown-pragmas -Wdeclaration-after-statement -fstrict-aliasing -O3 -DNDEBUG  -pipe -static-libgcc -static-libstdc++ contrib/dbs/CMakeFiles/tiff-grayscale.dir/tiff-grayscale.c.obj  -o contrib\dbs\tiff-grayscale.exe -Wl,--major-image-version,0,--minor-image-version,0  libtiff/libtiff.a port/libport.a C:/tools/cygwin64/lib/libm.a -Wl,-Bstatic -lz -ljpeg -llzma -lzstd -Wl,-Bdynamic -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
>
> D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmsvcrt.a(desis01151.o):(.text+0x0): multiple definition of `log10'
>
> C:/tools/cygwin64/lib/libm.a(t-d000863.o):fake:(.text+0x0): first defined here
>
> collect2.exe: error: ld returned 1 exit status
>
> ninja: build stopped: subcommand failed.
>
> build failed. Check D:/media-autobuild_suite-master/build/libtiff-git/build-64bit/ab-suite.build.log
>
> This is required for other packages, so this script will exit.
>
> 18:36:42   Creating diagnostics file...```
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jb-alvarado/media-autobuild_suite/issues/1162>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AH9XxoMiqReTiBtutEfFenET-mp_HpD4ks5vfM7zgaJpZM4clCk1>
> .
>

https://gist.githubusercontent.com/1480c1/3d981dd54aad0baeed8f822bb156fb68/raw/16ed09dafca3b4b646592c0cde82b122a643ced1/0001-Don-t-use-libm-if-MINGW-due-to-conflict-with-libmsvc.patch

This would be applied to libtiff.

in the Msys2 shell, get
https://gist.githubusercontent.com/1480c1/fe9d0ed8a3e353ff99397efe2ff85779/raw/68671f0414d83b1ab08e3a3ae016a70b260ac061/0001-Don-t-use-libm-with-libtiff-due-to-conflict-with-lib.patch
and
then run `git
apply 0001-Don-t-use-libm-with-libtiff-due-to-conflict-with-lib.patch`
This will make the suite apply the patch to libtiff automatically.

Unsure right now I should upstream since technically, there is no need to
link libm for mingw, however, probably will open a issue or pr with libtiff
later.


On Tue, Apr 9, 2019 at 12:50 PM Chris Degawa <cddegawa@gmail.com> wrote:

> That's due to a conflict with msvc's libs and mingw.
> I do have a patch for it, but was unsure if it was needed.
>
> On Tue, Apr 9, 2019 at 12:33 PM Hackerpcs <notifications@github.com>
> wrote:
>
>> Windows 10 LTSC 64 bit, Intel i5 2500k, GTX 1060 6GB (if it matters)
>>
>> logs.zip <https://0x0.st/zZYN.zip>
>>
>> Imported options from ffmpeg_options.txt
>>
>> Imported custom options from ffmpeg_options.txt
>>
>> License: nonfree
>>
>> Imported options from mpv_options.txt
>>
>>
>>
>>         Starting 64bit compilation of global tools
>>
>> 18:08:47 ├ Running download_rustup...
>>
>>
>>
>> [snip]
>>
>>
>>
>> 18:35:22 Installing libjpeg-turbo... done
>>
>> 18:35:26   Running git clone for libtiff...
>>
>> 18:35:46 ┌ libtiff git .............................. [Recently updated]
>>
>> 18:35:46 ├ Running cmake...
>>
>> 18:36:10 ├ Running build...
>>
>> Likely error:
>>
>>           ^~~~
>>
>> ninja: build stopped: subcommand failed.
>>
>> [1/6] Linking C executable contrib\addtiffo\addtiffo.exe
>>
>> [2/6] Linking C executable contrib\dbs\tiff-grayscale.exe
>>
>> FAILED: contrib/dbs/tiff-grayscale.exe
>>
>> cmd.exe /C "cd . && D:\media-autobuild_suite-master\msys64\mingw64\bin\gcc.exe -mthreads -mtune=generic -O2 -pipe -DLIBXML_STATIC -Wall -Winline -W -Wformat-security -Wpointer-arith -Wdisabled-optimization -Wno-unknown-pragmas -Wdeclaration-after-statement -fstrict-aliasing -O3 -DNDEBUG  -pipe -static-libgcc -static-libstdc++ contrib/dbs/CMakeFiles/tiff-grayscale.dir/tiff-grayscale.c.obj  -o contrib\dbs\tiff-grayscale.exe -Wl,--major-image-version,0,--minor-image-version,0  libtiff/libtiff.a port/libport.a C:/tools/cygwin64/lib/libm.a -Wl,-Bstatic -lz -ljpeg -llzma -lzstd -Wl,-Bdynamic -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
>>
>> D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmsvcrt.a(desis01151.o):(.text+0x0): multiple definition of `log10'
>>
>> C:/tools/cygwin64/lib/libm.a(t-d000863.o):fake:(.text+0x0): first defined here
>>
>> collect2.exe: error: ld returned 1 exit status
>>
>> ninja: build stopped: subcommand failed.
>>
>> build failed. Check D:/media-autobuild_suite-master/build/libtiff-git/build-64bit/ab-suite.build.log
>>
>> This is required for other packages, so this script will exit.
>>
>> 18:36:42   Creating diagnostics file...```
>>
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jb-alvarado/media-autobuild_suite/issues/1162>, or mute
>> the thread
>> <https://github.com/notifications/unsubscribe-auth/AH9XxoMiqReTiBtutEfFenET-mp_HpD4ks5vfM7zgaJpZM4clCk1>
>> .
>>
>

Theoretically, it should not matter mainly as libm should not be linked
since mingw-w64 gcc uses libmsvcrt.a for math by default. However, libtiff
has a system to autoinclude libm if it's found. If the patch is applied, it
will only add libm if the system is not mingw

On Tue, Apr 9, 2019 at 12:56 PM Chris Degawa <cddegawa@gmail.com> wrote:

>
> https://gist.githubusercontent.com/1480c1/3d981dd54aad0baeed8f822bb156fb68/raw/16ed09dafca3b4b646592c0cde82b122a643ced1/0001-Don-t-use-libm-if-MINGW-due-to-conflict-with-libmsvc.patch
>
> This would be applied to libtiff.
>
> in the Msys2 shell, get
> https://gist.githubusercontent.com/1480c1/fe9d0ed8a3e353ff99397efe2ff85779/raw/68671f0414d83b1ab08e3a3ae016a70b260ac061/0001-Don-t-use-libm-with-libtiff-due-to-conflict-with-lib.patch and
> then run `git
> apply 0001-Don-t-use-libm-with-libtiff-due-to-conflict-with-lib.patch`
> This will make the suite apply the patch to libtiff automatically.
>
> Unsure right now I should upstream since technically, there is no need to
> link libm for mingw, however, probably will open a issue or pr with libtiff
> later.
>
>
> On Tue, Apr 9, 2019 at 12:50 PM Chris Degawa <cddegawa@gmail.com> wrote:
>
>> That's due to a conflict with msvc's libs and mingw.
>> I do have a patch for it, but was unsure if it was needed.
>>
>> On Tue, Apr 9, 2019 at 12:33 PM Hackerpcs <notifications@github.com>
>> wrote:
>>
>>> Windows 10 LTSC 64 bit, Intel i5 2500k, GTX 1060 6GB (if it matters)
>>>
>>> logs.zip <https://0x0.st/zZYN.zip>
>>>
>>> Imported options from ffmpeg_options.txt
>>>
>>> Imported custom options from ffmpeg_options.txt
>>>
>>> License: nonfree
>>>
>>> Imported options from mpv_options.txt
>>>
>>>
>>>
>>>         Starting 64bit compilation of global tools
>>>
>>> 18:08:47 ├ Running download_rustup...
>>>
>>>
>>>
>>> [snip]
>>>
>>>
>>>
>>> 18:35:22 Installing libjpeg-turbo... done
>>>
>>> 18:35:26   Running git clone for libtiff...
>>>
>>> 18:35:46 ┌ libtiff git .............................. [Recently updated]
>>>
>>> 18:35:46 ├ Running cmake...
>>>
>>> 18:36:10 ├ Running build...
>>>
>>> Likely error:
>>>
>>>           ^~~~
>>>
>>> ninja: build stopped: subcommand failed.
>>>
>>> [1/6] Linking C executable contrib\addtiffo\addtiffo.exe
>>>
>>> [2/6] Linking C executable contrib\dbs\tiff-grayscale.exe
>>>
>>> FAILED: contrib/dbs/tiff-grayscale.exe
>>>
>>> cmd.exe /C "cd . && D:\media-autobuild_suite-master\msys64\mingw64\bin\gcc.exe -mthreads -mtune=generic -O2 -pipe -DLIBXML_STATIC -Wall -Winline -W -Wformat-security -Wpointer-arith -Wdisabled-optimization -Wno-unknown-pragmas -Wdeclaration-after-statement -fstrict-aliasing -O3 -DNDEBUG  -pipe -static-libgcc -static-libstdc++ contrib/dbs/CMakeFiles/tiff-grayscale.dir/tiff-grayscale.c.obj  -o contrib\dbs\tiff-grayscale.exe -Wl,--major-image-version,0,--minor-image-version,0  libtiff/libtiff.a port/libport.a C:/tools/cygwin64/lib/libm.a -Wl,-Bstatic -lz -ljpeg -llzma -lzstd -Wl,-Bdynamic -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
>>>
>>> D:/media-autobuild_suite-master/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libmsvcrt.a(desis01151.o):(.text+0x0): multiple definition of `log10'
>>>
>>> C:/tools/cygwin64/lib/libm.a(t-d000863.o):fake:(.text+0x0): first defined here
>>>
>>> collect2.exe: error: ld returned 1 exit status
>>>
>>> ninja: build stopped: subcommand failed.
>>>
>>> build failed. Check D:/media-autobuild_suite-master/build/libtiff-git/build-64bit/ab-suite.build.log
>>>
>>> This is required for other packages, so this script will exit.
>>>
>>> 18:36:42   Creating diagnostics file...```
>>>
>>>
>>> —
>>> You are receiving this because you are subscribed to this thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/jb-alvarado/media-autobuild_suite/issues/1162>, or mute
>>> the thread
>>> <https://github.com/notifications/unsubscribe-auth/AH9XxoMiqReTiBtutEfFenET-mp_HpD4ks5vfM7zgaJpZM4clCk1>
>>> .
>>>
>>

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -407,6 +407,7 @@ if enabled_any libwebp libtesseract &&
     do_vcs "https://gitlab.com/libtiff/libtiff.git"; then
     do_uninstall "${_check[@]}"
     sed -i 's/Libs.private.*/& -ljpeg -llzma -lz -lzstd/' libtiff-4.pc.in
+    do_patch "https://gitlab.com/libtiff/libtiff/merge_requests/73.patch"
     do_cmakeinstall -Dwebp=OFF -DUNIX=OFF
     do_checkIfExist
 fi

